physlr/physlr.py

```diff
@@ -967,15 +967,19 @@ def physlr_flesh_backbone(self):
 
     def physlr_subgraph(self):
         "Extract a vertex-induced subgraph."
-        if self.args.d not in (0, 1):
-            sys.exit("physlr subgraph: error: Only -d0 and -d1 are currently supported.")
+        if self.args.d > 50:
```

Comment:
50 seems to be a 'magic number' here?
It would be better for it to be a variable somewhere with default 50 I think -- easier to change in the future

Reply:
Actually 50 is a fixed limit I defined for `self.args.d` as larger values make it slow, and I don't see a usage for it for now; but I can remove this limitation. This is not an important parameter to be a variable...

Reply:
Having a 'magic number' isn't good coding practice. 
If we keep these references to the number, it was a number you decided on, but that doesn't mean we won't change it in the future, and I didn't understand why it was that particular value reading the code.

Reply:
Umm, you are totally right about not putting `magic numbers`, but the solution would be to totally remove this restriction on `self.args.d`... because the algorithm can theoretically to the job for any value of `self.args.d`... I mean there is no need for any if-statements on the parameter `self.args.d`. I am not against making that a variable, but now I think a fixed value or even a variable does not make that much sense here... the only problem is that it may be slow... but it will do the job... and I have already tested a slow case where it returned the whole graph as the subgraph, and it was fine.

Reply:
I think what Lauren meant is you should make it a parameter.

Reply:
I just added a parameter called 'self.args.d_max'